src/components/shared/Entity/DetailsForm/useDataPlaneField.ts

```diff
@@ -50,6 +53,15 @@ export default function useDataPlaneField(
 
                 dataPlanesOneOf.push({ const: option, title });
             });
+        } else {
+            // The details form store hydrator does not fail loudly when no data-plane options are found
+            // and the create workflow does not have a fallback data-plane option to use in that scenario.
+            dataPlanesOneOf.push({
+                const: DATA_PLANE_OPTION_TEMPLATE,
+                title: intl.formatMessage({
+                    id: 'workflows.dataPlane.label.noOptionsFound',
+                }),
+            });
```

Comment:
So adding this will mean the initial value is no right and will switch the autocomplete between `controlled` and `uncontrolled`. We'll need to make sure either the renderer handles this and knows when to default the value or we set the value ahead of time.